src/main/java/com/pinterest/secor/consumer/Consumer.java

```diff
@@ -267,4 +243,36 @@ protected boolean consumeNextMessage() {
     public OffsetTracker getOffsetTracker() {
         return this.mOffsetTracker;
     }
+
+    private void adjustOffsets(Message rawMessage) {
+        try {
+            mMessageWriter.adjustOffset(rawMessage, isLegacyConsumer);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to adjust offset.", e);
+        }
+    }
+
+    private void writeMessage(Message rawMessage, ParsedMessage parsedMessage) throws Exception {
+        mMessageWriter.write(parsedMessage);
+        mBadMessages *= DECAY;
```

Comment:
`DECAY` is final and used to decrease `mBadMessages` as before. It does not make sense to declare new `DECAY` variable on each `writeMessage` call, so I moved it to static final constant.